From 1d74b943135776b430424e209ccc89c5e561828b Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Thu, 30 Dec 2021 07:32:48 +0200
Subject: [PATCH] [SW-68384] gaudi2: security: set axprot for psoc jtag

By default the permission level of jtag mstr if is
privilege therefore any transaction coming via jtag
is treated as privilege transaction. hence setting the
permission level to user level.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: If902a8a93aea7ca1d97afae1e0d659f1989a4f24
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index cb9d03c..6ff12ae 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -44,6 +44,7 @@
 #include <dcore0_vdec0_mstr_if_core_hbw_regs.h>
 #include <dcore0_vdec0_mstr_if_core_lbw_regs.h>
 #include <pmmu_hbw_mstr_if_core_hbw_regs.h>
+#include <jt_mstr_if_core_hbw_regs.h>
 #include "common.h"
 
 static struct hl_sec_rrs_cfg hl_sec_lbw_rrs_tbl[] = {
@@ -1137,6 +1138,12 @@
 	regmap_writel(HL_SEC_AXPROT_SEC_VAL, mmPMMU_HBW_MSTR_IF_CORE_HBW_AXPROT_OVRD);
 }
 
+static void preboot_secure_psoc(void)
+{
+	/* Set jtag to user level */
+	regmap_writel(HL_SEC_AXPROT_USER_VAL, mmJT_MSTR_IF_CORE_HBW_AXPROT_OVRD);
+}
+
 void preboot_security_init(void)
 {
 	preboot_secure_pcie();
@@ -1147,6 +1154,7 @@
 		preboot_lbw_rtr_preconfig();
 		hl_dev_prot_bits_init();
 		preboot_secure_pmmu();
+		preboot_secure_psoc();
 		preboot_secure_lbw();
 		preboot_secure_hbw();
 		preboot_puts("Security enabled.\n");
